### PR TITLE
Themeable Switch thumb images (active/inactive)

### LINKED ISSUE #81564: Themeable thumb images for `Switch`
Add `activeThumbImage` and `inactiveThumbImage` to `SwitchThemeData`

### PR DESCRIPTION
**Pull request by kirolous-nashaat**
Adding `SwitchThemeData.activeThumbImage` & `SwitchThemeData.inactiveThumbImage`

Fixes [#81564](https://github.com/flutter/flutter/issues/81564) 

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

**Comment by kirolous-nashaat** [comment]
I couldn't find a meaningful way to test the usage of these parameters as default images for the switch. neither adding them to `debugFillProperties` because of the reference value (i.e: #a1231).

**Comment by HansMuller** [comment]
This looks like a situation that would be better served by a MaterialStateProperty

File: packages/flutter/lib/src/material/switch.dart [hunk 1]
```diff
@@ -281,6 +283,41 @@ class Switch extends StatelessWidget {
...
+  /// {@tool snippet}
+  /// This example resolves the [thumbImage] based on the current
+  /// [MaterialState] of the [Switch], providing a different [ImageProvider] when it is
+  /// [MaterialState.disabled].
+  ///
+  /// ```dart
+  /// Switch(
+  ///   value: true,
+  ///   onChanged: (_) => true,
+  ///   thumbColor: MaterialStateProperty.resolveWith<ImageProvider>((Set<MaterialState> states) {
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/switch.dart:304]: I think you want a thumbImage example here

**Comment by kirolous-nashaat** [comment]
> This looks like a situation that would be better served by a MaterialStateProperty

Updated the PR to use MaterialStateProperty.
but I don't know how to test that the imageprovider values are being used in the `_SwitchPainter`, Could you please point me to an example for this ?

File: packages/flutter/lib/src/material/switch.dart (hunk 1 above)
Comment (kirolous-nashaat) [reply packages/flutter/lib/src/material/switch.dart:304]: Done ✔️

**Comment by QuncCccccc** [comment]
Hi! Thanks so much for your contribution, I'm closing this draft since there's some duplicate work ([PR](https://github.com/flutter/flutter/pull/110095)). Thanks again. Please let me know if there are any questions!

**Pull request closed without merging**